File: servers/scalatra/src/main/scala/com/wordnik/client/model/position.scala
package com.wordnik.client.model

import java.util.Date
case class Position (
  account: Double,

  symbol: String,

  underlying: Option[String],

  quoteCurrency: Option[String],

  currency: Option[String],

  commission: Option[Double],

  prevRealisedPnl: Option[Double],

  prevUnrealisedPnl: Option[Double],

  prevClosePrice: Option[Double],

  realisedCost: Option[Double],

  realisedPnl: Option[Double],

  openingTimestamp: Option[Date],

  openingQty: Option[Double],

  openingCost: Option[Double],

  openingComm: Option[Double],

  openOrderBuyQty: Option[Double],

  openOrderBuyCost: Option[Double],

  openOrderSellQty: Option[Double],

  openOrderSellCost: Option[Double],

  execBuyQty: Option[Double],

  execBuyCost: Option[Double],

  execSellQty: Option[Double],

  execSellCost: Option[Double],

  execQty: Option[Double],

  execCost: Option[Double],

  execComm: Option[Double],

  currentTimestamp: Option[Date],

  currentQty: Option[Double],

  currentCost: Option[Double],

  currentComm: Option[Double],

  unrealisedCost: Option[Double],

  grossOpenCost: Option[Double],

  grossExecCost: Option[Double],

  lastPrice: Option[Double],

  lastValue: Option[Double],

  homeNotional: Option[Double],

  foreignNotional: Option[Double],

  notionalCurrency1: Option[String],

  notionalValue1: Option[Double],

  notionalCurrency2: Option[String],

  notionalValue2: Option[Double],

  initMargin: Option[Double],

  maintMargin: Option[Double],

  sessionMargin: Option[Double],

  targetExcessMargin: Option[Double],

  varMargin: Option[Double],

  unrealisedPnl: Option[Double],

  unrealisedPnlPcnt: Option[Double],

  avgEntryPrice: Option[Double],

  breakEvenPrice: Option[Double],

  marginCallPrice: Option[Double],

  liquidationPrice: Option[Double],

  bankruptPrice: Option[Double],

  timestamp: Option[Date]

  )

File: clients/typescript-angular/API/Client/Position.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    export interface Position {

        account: number;

        symbol: string;

        underlying?: string;

        quoteCurrency?: string;

        currency?: string;

        commission?: number;

        crossMargin?: boolean;

        rebalancedPnl?: number;

        prevRealisedPnl?: number;

        prevUnrealisedPnl?: number;

        prevClosePrice?: number;

        openingTimestamp?: date;

        openingQty?: number;

        openingCost?: number;

        openingComm?: number;

        openOrderBuyQty?: number;

        openOrderBuyCost?: number;

        openOrderBuyPremium?: number;

        openOrderSellQty?: number;

        openOrderSellCost?: number;

        openOrderSellPremium?: number;

        execBuyQty?: number;

        execBuyCost?: number;

        execSellQty?: number;

        execSellCost?: number;

        execQty?: number;

        execCost?: number;

        execComm?: number;

        currentTimestamp?: date;

        currentQty?: number;

        currentCost?: number;

        currentComm?: number;

        realisedCost?: number;

        unrealisedCost?: number;

        grossOpenCost?: number;

        grossOpenPremium?: number;

        grossExecCost?: number;

        isOpen?: boolean;

        markPrice?: number;

        markValue?: number;

        homeNotional?: number;

        foreignNotional?: number;

        posState?: string;

        posCost?: number;

        posCost2?: number;

        posCross?: number;

        posInit?: number;

        posComm?: number;

        posLoss?: number;

        posMargin?: number;

        posMaint?: number;

        posAllowance?: number;

        taxableMargin?: number;

        initMargin?: number;

        maintMargin?: number;

        sessionMargin?: number;

        targetExcessMargin?: number;

        varMargin?: number;

        realisedGrossPnl?: number;

        realisedTax?: number;

        realisedPnl?: number;

        unrealisedGrossPnl?: number;

        longBankrupt?: number;

        shortBankrupt?: number;

        taxBase?: number;

        indicativeTaxRate?: number;

        indicativeTax?: number;

        unrealisedTax?: number;

        unrealisedPnl?: number;

        unrealisedPnlPcnt?: number;

        unrealisedRoePcnt?: number;

        simpleQty?: number;

        simpleCost?: number;

        simpleValue?: number;

        simplePnl?: number;

        simplePnlPcnt?: number;

        avgCostPrice?: number;

        avgEntryPrice?: number;

        breakEvenPrice?: number;

        marginCallPrice?: number;

        liquidationPrice?: number;

        bankruptPrice?: number;

        timestamp?: date;
    }

}
